Src/Support/Google.Apis.Auth/OAuth2/IGoogleCredential.cs

```diff
@@ -55,6 +57,36 @@ internal interface IGoogleCredential : ICredential, ITokenAccessWithHeaders, IHt
         /// </summary>
         bool SupportsExplicitScopes { get; }
 
+        /// <summary>
+        /// Returns the universe domain this credential belongs to.
+        /// </summary>
+        /// <remarks>
+        /// For most credential types, this operation is synchronous and will always
+        /// return a completed task.
+        /// For <see cref="ComputeCredential"/>, the universe domain is obtained from the
+        /// metadata server, which requires an HTTP call. This value is obtained only once,
+        /// the first time it is requested for any instance of <see cref="ComputeCredential"/>.
+        /// After that, this method will always return a completed task.
+        /// The task's result will never be null.
+        /// </remarks>
+        Task<string> GetUniverseDomainAsync(CancellationToken cancellationToken);
```

Comment:
And presumably the cancellation token should only ever apply to this call - if this call times out, a later call could still succeed?